fxa-oauth-server/test/api.js

```diff
@@ -2699,27 +2699,27 @@ describe('/v1', function() {
           });
       });
 
-      it('succeeds for clients that do not have the scope, returning an empty object', () => {
+      it('fails for clients that are not allowed the requested scope', () => {
```

Comment:
Switching to the shared logic has changed the behaviour of `/key-data` in the case where the client is not allowed one of the requested scopes.  Previously it would just ignore that scope and not return its corresponding key data, while now it throws a hard error.  I submit that this is a change for the better, as the flow was never going to succeed anyway, so we might as well error out as early as possible.

Reply:
👍

Reply:
Do we have to worry about any clients depending on the previous behavior?